File: typescript/src/TopInterview150/Array-String/MajorityElement/MajorityElement.test.ts
import {majorityElement} from './MajorityElement';

describe('Majority Element', () => {
  it('example 1', () => {
    expect(majorityElement([3, 2, 3])).toBe(3);
  });
  it('example 2', () => {
    expect(majorityElement([2, 2, 1, 1, 1, 2, 2])).toBe(2);
  });
  it('length 1', () => {
    expect(majorityElement([0])).toBe(0);
  });
  it('length 2', () => {
    expect(majorityElement([0, 0])).toBe(0);
  });
  it('length 3: mode at beginning', () => {
    expect(majorityElement([0, 0, 1])).toBe(0);
  });
  it('length 3: mode at end', () => {
    expect(majorityElement([0, 1, 1])).toBe(1);
  });
  it('length 3: all same element', () => {
    expect(majorityElement([1, 1, 1])).toBe(1);
  });
  it('length 4: beginning', () => {
    expect(majorityElement([2, 2, 2, 1])).toBe(2);
  });
  it('length 4: end', () => {
    expect(majorityElement([1, 2, 2, 2])).toBe(2);
  });
  it('length 4: all same', () => {
    expect(majorityElement([3, 3, 3, 3])).toBe(3);
  });
  it('length 5', () => {
    expect(majorityElement([5, 3, 3, 1, 3])).toBe(3);
  });
  it('oops', () => {
    expect(majorityElement([4, 4, 3, 3, 4, 4, 2])).toBe(4);
  });
});

File: scala/src/test/scala/jbaldassari/leetcode/topinterview150/arraystring/majorityelement/MajorityElementTest.scala
package jbaldassari.leetcode.topinterview150.arraystring.majorityelement

class MajorityElementTest extends munit.FunSuite {
  test("example 1") {
    assertEquals(Solution.majorityElement(Array(3, 2, 3)), 3)
  }
  test("example 2") {
    assertEquals(Solution.majorityElement(Array(2, 2, 1, 1, 1, 2, 2)), 2)
  }
  test("length 1") {
    assertEquals(Solution.majorityElement(Array(0)), 0)
  }
  test("length 2") {
    assertEquals(Solution.majorityElement(Array(0, 0)), 0)
  }
  test("length 3: majority at beginning") {
    assertEquals(Solution.majorityElement(Array(0, 0, 1)), 0)
  }
  test("length 3: majority at end") {
    assertEquals(Solution.majorityElement(Array(0, 1, 1)), 1)
  }
  test("length 3: all same element") {
    assertEquals(Solution.majorityElement(Array(1, 1, 1)), 1)
  }
  test("length 4: beginning") {
    assertEquals(Solution.majorityElement(Array(2, 2, 2, 1)), 2)
  }
  test("length 4: end") {
    assertEquals(Solution.majorityElement(Array(1, 2, 2, 2)), 2)
  }
  test("length 4: all same") {
    assertEquals(Solution.majorityElement(Array(3, 3, 3, 3)), 3)
  }
  test("length 5") {
    assertEquals(Solution.majorityElement(Array(5, 3, 3, 1, 3)), 3)
  }
  test("length 7") {
    assertEquals(Solution.majorityElement(Array(4, 4, 3, 3, 4, 4, 2)), 4)
  }
}
